src/Mvc/Mvc.Abstractions/src/ActionContext.cs

```diff
@@ -31,10 +31,10 @@ public ActionContext()
         /// <param name="actionContext">The <see cref="ActionContext"/> to copy.</param>
         public ActionContext(ActionContext actionContext)
             : this(
-                actionContext?.HttpContext,
-                actionContext?.RouteData,
-                actionContext?.ActionDescriptor,
-                actionContext?.ModelState)
+                actionContext?.HttpContext!,
+                actionContext?.RouteData!,
+                actionContext?.ActionDescriptor!,
+                actionContext?.ModelState!)
         {
```

Comment:
I'm missing something, how can actionContext be null here if you are requiring it not to be null? Why do we need the `?` operators and the `!`?

Reply:
could we do actionContext! instead here?

Reply:
for the other constructor we should remove the nullchecks and use the `!` in the parameter instead.

Reply:
You're right, i'll change the paramater to be nullable. There are unit test code paths this gets used in where the context is null